app/models/agents/s3_agent.rb

```diff
@@ -234,7 +234,12 @@ def emit_event?(event_type)
 
     def process_file?(file_name)
       return true unless filename_filter_present?
-      prefix_match?(file_name) || suffix_match?(file_name)
+
+      if both_filename_filters_present?
+        prefix_match?(file_name) && suffix_match?(file_name)
```

Comment:
Not in love with this approach. Shouldn't both methods only return false if they are present and failing? And then we just always use `&&`?

Reply:
You're right it was overcomplicated.